### PR TITLE
HOTT-1009: Adds step to take input for the meursing additional code

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

https://transformuk.atlassian.net/browse/HOTT-1009

### What?

Screenshot from 2021-11-01 17-38-36

I have added/removed/altered:

- [x] Added step to take input for the meursing additional code
- [x] Make sure the step fits into the row to ni and gb to ni journeys
- [x] Updated certificate of origin step to go to interstitial page with a negative answer
- [x] Updated interstitial page to include the certificate of origin variation of the interstitial message

### Why?

I am doing this because:

- This is needed so we can resolve measures that have measure components with meursing duty expressions and then calculate duties for them

### CODE REVIEW

File: app/views/steps/interstitial/show.html.erb [hunk 1]
```diff
@@ -3,7 +3,11 @@
 <span class="govuk-caption-xl">Calculate import duties</span>
 
 <% if user_session.gb_to_ni_route? %>
-  <%= render partial: 'steps/interstitial/shared/context', locals: { heading: t('interstitial.gb_to_ni.heading'), body: t('interstitial.gb_to_ni.body') } %>
+  <% if user_session.trade_defence? %>
```
Comment (willfish) [inline app/views/steps/interstitial/show.html.erb:6]: I'm going to simplify this with an interstitial helper: https://transformuk.atlassian.net/browse/HOTT-1083.

A view shouldn't have this much logic in it IMO

File: app/controllers/steps/meursing_additional_codes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    end
+
+    private
+
+    def meursing_additional_code_params
+      if params[:steps_meursing_additional_code].present?
+        params.require(:steps_meursing_additional_code).permit(:meursing_additional_code)
+      else
+        {}
+      end
```
Comment (alexdesi) [inline app/controllers/steps/meursing_additional_codes_controller.rb:20]: I think you can use fetch here, something like:
`params.fetch(foo:, {}).permit(:bar, :baz)`

(found on https://stackoverflow.com/questions/17687506/how-to-make-an-optional-strong-parameters-key-but-filter-nested-params)

File: app/helpers/meursing_additional_code_helper.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    hint_text = t(
+      'meursing_additional_code.hint_text',
+      link: meursing_lookup_link,
+    )
+
+    sanitize(
+      hint_text,
+      tags: %w[p a],
+      attributes: %w[class target rel href],
+    )
```
Comment (alexdesi) [inline app/helpers/meursing_additional_code_helper.rb:12]: I am not sure if this is used?

**Review by alexdesi — APPROVED** [review]
✅

**Pull request merged**